src/ImageSharp.Web/Commands/CommandCollection.cs

```diff
@@ -29,16 +29,7 @@ private CommandCollection(IEqualityComparer<string> comparer)
         /// <summary>
         /// Gets an <see cref="IEnumerable{String}"/> representing the keys of the collection.
         /// </summary>
-        public IEnumerable<string> Keys
-        {
-            get
-            {
-                foreach (KeyValuePair<string, string> item in this)
-                {
-                    yield return this.GetKeyForItem(item);
-                }
-            }
-        }
+        public IEnumerable<string> Keys => this.Dictionary.Keys; // TODO Change return type to ICollection<string> in next major release
```

Comment:
This is wrong and unrelated. Please revert.

Reply:
I also noticed the failing test and reverted this change, but using `Dictionary.Keys` does prevent allocating a new list.

I've updated my branch to fix this, expose it as an internal `AllKeys` property and added [benchmarks](https://github.com/ronaldbarendse/ImageSharp.Web/blob/f398b7f5bccd6bcd7abde3f73059cefb80b1554b/tests/ImageSharp.Web.Benchmarks/Commands/CommandCollectionBenchmarks.cs):

```
BenchmarkDotNet=v0.13.1, OS=Windows 10.0.22000
Intel Core i7-10750H CPU 2.60GHz, 1 CPU, 12 logical and 6 physical cores
.NET SDK=6.0.202
    [Host]     : .NET 5.0.16 (5.0.1622.16705), X64 RyuJIT
    DefaultJob : .NET 5.0.16 (5.0.1622.16705), X64 RyuJIT


|   Method |      Mean |    Error |   StdDev | Ratio | RatioSD |  Gen 0 | Allocated |
|--------- |----------:|---------:|---------:|------:|--------:|-------:|----------:|
|     Keys |  87.77 ns | 1.560 ns | 2.136 ns |  1.00 |    0.00 | 0.0153 |      96 B |
| KeysList | 152.28 ns | 1.699 ns | 1.589 ns |  1.73 |    0.04 | 0.0355 |     224 B |
|  AllKeys |  37.82 ns | 0.637 ns | 0.806 ns |  0.43 |    0.02 | 0.0063 |      40 B |
```

Reply:
If you have an improvement it is welcome but it must be part of a separate PR.